ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/interceptor/AuthorizationInterceptor.scala

```diff
@@ -47,6 +49,14 @@ final class AuthorizationInterceptor(protected val authService: AuthService, ec:
       FutureConverters
         .toScala(authService.decodeMetadata(headers))
         .onComplete {
+          case Failure(exception) =>
+            logger.warn(s"Failed to get claims from request metadata: ${exception.getMessage}")
+            call.close(internalAuthenticationError, new Metadata())
+            new ServerCall.Listener[Nothing]() {}
+          case Success(Claims.empty) =>
+            logger.debug(s"Auth metadata decoded into empty claims, returning UNAUTHENTICATED")
+            call.close(Status.UNAUTHENTICATED, new Metadata())
+            new ServerCall.Listener[Nothing]() {}
```

Comment:
This assumes that the AuthService returns empty claims if the request is not authenticated (missing/invalid token). That leaves the AuthService no way to signal that the user was authenticated, but has no authorization (not even for the public services).

This is not a big issue, but we should add a corresponding comment to the docs of `AuthService.decodeMetadata`.

Reply:
I thought you needed at least `ClaimPublic` to access public services.

Reply:
If that wasn't the case then I have no idea why the tests are green. :smile:

Reply:
> I thought you needed at least ClaimPublic to access public services.

That is correct. The sandbox AuthService will always include at least a `ClaimPublic` for any valid token, but this might not be the case for other AuthService implementations.

That's why I thought we should add a note at rc/main/scala/com/digitalasset/ledger/api/auth/AuthService.scala#L26, something like:

```scala
/**
Return empty [[Claims]] to reject requests with a UNAUTHENTICATED error status.
Return [[Claims]] with only a single [[ClaimPublic]] claim to reject all non-public requests with a PERMISSION_DENIED status.
Return a failed future to reject requests with an INTERNAL error status.
*/
```

Reply:
Will do, thanks!

Reply:
Addressed by https://github.com/digital-asset/daml/pull/4485/commits/90490b862b1cef7095149d345f5792399fe9ce05
Thank you for the clarification!